File: CodingTest/프로그래머스/Level_1/신고_결과_받기.swift
import Foundation

func solution(_ id_list:[String], _ report:[String], _ k:Int) -> [Int] {
  var reported : [String : Int] = [:]
  var user : [String : [String]] = [:]
  var reportedName : [String] = []

  for r in Set(report) {
    let splited = r.split(separator: " ").map{String($0)}
    user[splited[0]] = (user[splited[0]] ?? []) + [splited[1]]
    reported[splited[1]] = (reported[splited[1]] ?? 0) + 1
    }
  print(user)
  print(reported)
  for i in reported{
    if i.value >=  k {
      reportedName.append(i.key)
    }
  }
  print(reportedName)

  return id_list.map { id in
      return (user[id] ?? []).reduce(0) {
          $0 + ((reported[$1] ?? 0) >= k ? 1 : 0)
      }
  }
}


print(solution(["muzi", "frodo", "apeach", "neo"], ["muzi frodo","apeach frodo","frodo neo","muzi neo","apeach muzi"], 2))

File: CodingTest/프로그래머스/Level_1/python/신고_결과_받기.py
from collections import defaultdict

def solution(id_list, report, k):
    answer = []

    x = defaultdict(set)
    y = defaultdict(set)
    for i in report:
        report_from, report_to = i.split(' ')
        x[report_from].add(report_to)
        y[report_to].add(report_from)
    
    for id in id_list:
        count = 0 
        for re in x[id]:
            if len(y[re]) >= k:
                count += 1
        answer.append(count)
    return answer